src/test/java/HttpServletResponseFake.java

```diff
@@ -0,0 +1,224 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Collection;
+import java.util.Locale;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletResponse;
+
+/** Fake version of HttpServletResponse used for testing */
+public class HttpServletResponseFake implements HttpServletResponse {
+  // Stores the status code of the http request (default to 200 until error sent)
+  private int responseCode = 200;
+
+  // Used for storing and returning the passed JSON response
+  private StringWriter stringWriter = new StringWriter();
+  private PrintWriter printWriter = new PrintWriter(stringWriter);
+
+  /**
+   * Get the stringwriter used for writing the response
+   *
+   * @return Stringwriter object
+   */
+  public StringWriter getStringWriter() {
+    printWriter.flush();
+    return stringWriter;
+  }
+
+  @Override
```

Comment:
Please leave all the unsupported methods together.

Reply:
Done! Thanks!